### PR TITLE
add missing scaleset nodes

### PR DESCRIPTION
**Pull request by bmc-msft**
Scalesets can have nodes that never check in (such as broken OS setup scripts).

This will add nodes that Azure knows about but have not checked in such that the `dead node` detection will eventually reimage the node.

NOTE: If node setup takes longer than NODE_EXPIRATION_TIME (1 hour), this will cause the nodes to continuously get reimaged.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/workers/nodes.py [hunk 1]
```diff
@@ -49,21 +49,25 @@ def create(
...
-        node.save()
-        send_event(
-            EventNodeCreated(
-                machine_id=node.machine_id,
-                scaleset_id=node.scaleset_id,
-                pool_name=node.pool_name,
+        # `save` only returns None when `new` is True and an object already
+        # exists in Azure Tables. If we're in that case, don't send an event.
+        result = node.save(new=new)
+        if result is None:
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/workers/nodes.py:63]: Per the comment, did you want:
```suggestion
        if result is not None:
```

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py [hunk 2]
```diff
@@ -300,6 +301,35 @@ def cleanup_nodes(self) -> bool:
                 )
                 node.delete()
 
+        # Scalesets can have nodes that never check in (such as broken OS setup
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/workers/scalesets.py:304]: 👍 helpful explanation, thanks.

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py [hunk 3]
```diff
@@ -300,6 +301,35 @@ def cleanup_nodes(self) -> bool:
...
+            if machine_id in node_machine_ids:
+                continue
+
+            logging.info(
+                "scaleset - adding missing azure node: %s:%s",
+                self.scaleset_id,
+                machine_id,
+            )
+
+            # Note, using `new=True` makes it such that if a node already has
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/workers/scalesets.py:323]: 👍

File: src/api-service/__app__/onefuzzlib/workers/nodes.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/workers/nodes.py:64]: The comment was worded oddly which lead to the misinterpretation.   Then I clicked the wrong button in github, which accepted this suggestion.  I reverted the suggestion and updated the comment.

**Review by ranweiler — APPROVED** [review]

**Pull request merged**